src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -240,6 +264,54 @@ ZoneSet::MoveWindowIntoZoneByIndex(HWND window, HWND windowZone, int index) noex
     }
 }
 
+IFACEMETHODIMP_(void)
+ZoneSet::MoveWindowIntoZoneByIndexSet(HWND window, HWND windowZone, const std::vector<int>& indexSet) noexcept
+{
+    if (m_zones.empty())
+    {
+        return;
+    }
+
+    while (auto zoneDrop = ZoneFromWindow(window))
+    {
+        zoneDrop->RemoveWindowFromZone(window, !IsZoomed(window));
+    }
```

Comment:
I think that block should be moved after 280-284

Reply:
Absolutely not, doing that would remove a window from the zone to which we just added it, in case we're snapping to a single zone.